### ISSUE TITLE
Static build results in runtime error finding librdpsnd-client.so

### ISSUE THREAD

**Issue by RickyDoug** [FreeRDP/FreeRDP#7779]
Version 2.4.1, from git clone, building on Centos 7.

Using -DBUILD_SHARED_LIBS=OFF results in static libraries being built, however when trying to run the client, it fails while trying to load a dynamic library it never built and should not be configured for. Since I also have disabled sound it should never be looking for rdpsnd.
-DCHANNEL_RDPSND=OFF -DCHANNEL_RDPSND_CLIENT=OFF -DCHANNEL_RDPSND_SERVER=OFF

During build, I get Linking C static library xxx.a as I expect.

But running the client, it attempts to load librdpsnd-client.so.

LoadLibraryA failed with /usr/local/lib64/freerdp2/librdpsnd-client.so: cannot open shared object file: No such file or directory.

**Comment by akallabeth** [comment]
1. Please ensure you´re calling the correct binary
1. Please always start with a clean source/build directory.

I´ve just build `2.4.1`, `2.6.1` and current `master` and none of them call `LoadLibrary` when connecting to a windows 10 (sound enabled)

**Comment by RickyDoug** [comment]
I just commented out the bit where it calls the plugins and it works just fine:)  I connect to a Windows RDP and all is good.

But why would a static build EVER be looking for a dll?

**Comment by akallabeth** [comment]
@RickyDoug 
* `rdpsnd` is a core component of the RDP protocol (which is required for almost everything aside legacy RDP operations) - you seem to have very low requirements for your client ;)
* What does module (plugin) loading have to do with `static` or `dynamic` builds?

**Comment by RickyDoug** [comment]
The module plugin loading appears to be broken if you statically build. I would expect static builds to statically compile the plugins and be built into the binary. This is broken in this case. 

'plugins' would have to exist at compile time, and they would be 'plugged in' at that time as well. You get compile time error checking and not run time error checking.

I would expect static builds to be a stand alone binary, fully self sufficient aside from the run time loader.
There are 2 reasons for this:
1) You eliminate a run time failure mode for the client. One could make the case that since it is windows, you always have a high chance of runtime failure.
2) You eliminate a 'complicated' install, with potential permissions issues on some platforms. 

But if this works for everyone else, I have a perfectly acceptable workaround by simply not loading plugins.

**Comment by akallabeth** [comment]
@RickyDoug you seem to miss the point.
* `rdpsnd` is an essential component that is required for full functionality of `CORE` `RDP` protocol. 
* static builds compile in everything you have activated and fall back to library search paths if some essential stuff is not found (your case)
* that it works for you without it is a miracle (as in you don´t use anything of RDP, or have not used anything yet, because everything after `WinXP` era requires `rdpsnd`, read gfx/clipboard/smartcard/drive redirection/... to name a few)

so, long story short, you are in unexplored lands, don´t complain if stuff explodes in your face ;)

**Comment by RickyDoug** [comment]
freerdp was the first RDP client we've looked at, but you just made me realize it is not a good fit for our use. Thanks for pointing that out! I've found several other implementations that are much cleaner and thankfully do not use cmake. 
Appreciate your help!
